vowpalwabbit/log_multi.cc

```diff
@@ -172,7 +172,10 @@ void display_tree_dfs(log_multi& b, const node& node, uint32_t depth)
 
 bool children(log_multi& b, uint32_t& current, uint32_t& class_index, uint32_t label)
 {
-  class_index = static_cast<uint32_t>(b.nodes[current].preds.unique_add_sorted(node_pred(label)));
+  auto& preds = b.nodes[current].preds;
```

Comment:
it looks like `unique_add_sorted` will only add the value if it doesn't already exist in the vector

Reply:
lower_bound gives you the behavior you want without having to do the iteration subtraction
Just make sure you check if the returned iterator is end() before you try to dereference it

```
found_it = lower_bound();
if(found_it != vec.end() && *found_it != val_to_insert) { vec.insert(); }
```

Reply:
This requires calling both lower and upper bound though

```
found_it = lower_bound();
if(found_it == vec.end() || *found_it != val_to_insert) { vec.insert(upper_bound); }
```

Reply:
upper_bound gives you strictly > while lower_bound is >=. So if the value doesn't exist, then it gives you exactly the same result as an upper_bound call would give

Reply:
Ah I see, that makes sense